src/etc/tidy.py

```diff
@@ -43,78 +43,184 @@ def do_license_check(name, contents):
     if not check_license(name, contents):
         report_error_name_no(name, 1, "incorrect license")
 
-
-file_names = [s for s in sys.argv[1:] if (not s.endswith("_gen.rs"))
-                                     and (not ".#" in s)]
-
 current_name = ""
 current_contents = ""
 check_tab = True
 check_cr = True
 check_linelength = True
 
+if len(sys.argv) < 2:
+    print "usage: tidy.py <src-dir>"
+    sys.exit(1)
+
+src_dir = sys.argv[1]
 
 try:
-    for line in fileinput.input(file_names,
+    count_rs = 0
+    count_py = 0
+    count_js = 0
+    count_sh = 0
+    count_pl = 0
+    count_c = 0
+    count_h = 0
+    count_other = 0
+
+    count_lines = 0
+    count_non_blank_lines = 0
+
+    def update_counts(current_name):
+        global count_rs
+        global count_py
+        global count_js
+        global count_sh
+        global count_pl
+        global count_c
+        global count_h
+        global count_other
+
+        if current_name.endswith(".rs"):
+            count_rs += 1
+        if current_name.endswith(".py"):
+            count_py += 1
+        if current_name.endswith(".js"):
+            count_js += 1
+        if current_name.endswith(".sh"):
+            count_sh += 1
+        if current_name.endswith(".pl"):
+            count_pl += 1
+        if current_name.endswith(".c"):
+            count_c += 1
+        if current_name.endswith(".h"):
+            count_h += 1
+
+    all_paths = set()
+
+    for (dirpath, dirnames, filenames) in os.walk(src_dir):
+
+        # Skip some third-party directories
```

Comment:
Similarly, we could probably define a set/list of skippable directories near the start of the file.

```python
skippable_dirs = {
    'src/jemalloc',
    'src/llvm',
    # ...
    'src/rust-installer'
}

if any(d in dirpath for d in skippable_dirs):
    continue